src/ethereum/cancun/blocks.py

```diff
@@ -98,3 +105,28 @@ class Receipt:
     cumulative_gas_used: Uint
     bloom: Bloom
     logs: Tuple[Log, ...]
+
+
+def encode_receipt(tx: Transaction, receipt: Receipt) -> Union[Bytes, Receipt]:
+    """
+    Encodes a receipt.
+    """
+    if isinstance(tx, AccessListTransaction):
+        return b"\x01" + rlp.encode(receipt)
+    elif isinstance(tx, FeeMarketTransaction):
+        return b"\x02" + rlp.encode(receipt)
+    elif isinstance(tx, BlobTransaction):
+        return b"\x03" + rlp.encode(receipt)
+    else:
+        return receipt
+
+
+def decode_receipt(receipt: Union[Bytes, Receipt]) -> Receipt:
+    """
+    Decodes a receipt.
+    """
+    if isinstance(receipt, Bytes):
+        assert receipt[0] in (1, 2, 3, 4)
```

Comment:
Confirming that this is supposed to include `4` but `encode_receipt` doesn't encode with a `4`.

Reply:
Ah. Yeah. Fixed for cancun and berlin